atlasdb-client/src/main/java/com/palantir/atlasdb/util/AccumulatingValueMetric.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.util;
+
+public class AccumulatingValueMetric extends CurrentValueMetric<Long> {
+    private CurrentValueMetric<Long> delegate = new CurrentValueMetric<>();
+
+    public AccumulatingValueMetric() {
+        setValue(0L);
+    }
+
+    @Override
+    public Long getValue() {
+        return delegate.getValue();
+    }
+
+    @Override
+    public void setValue(Long newValue) {
+        delegate.setValue(newValue);
+    }
+
+    public void accumulateValue(Long newValue) {
+        setValue(newValue + getValue());
```

Comment:
This is not thread safe.  Is it guaranteed that this will only be called in a single thread?  Does this guarantee still hold if Nimbus is sweeping many products at once?

We should either make this safe or make it very clear to our future selves that this is an issue we need to fix.